design-system/src/theme-provider.tsx

```diff
@@ -28,31 +27,35 @@ type ThemeProviderProps = {
 };
 
 const ThemeProvider = (props: ThemeProviderProps) => {
-  const root = useRef<HTMLDivElement>();
+  const root = document.querySelector(':root') as HTMLElement;
```

Comment:
Although the we are not expecting the `ThemeProvider` to be re-rendered many times, perhaps we can keep using a ref here as the root element (`html`) will not change and so we don't need to query the document on every render.

Reply:
I'm sorry but I don't know if I follow it ([considering the previous comment](https://github.com/commercetools/ui-kit/pull/2227#discussion_r930015715)). Is your point here to use `forwardRef` for this component and expect the ref to `root` element be provided in the client code?

Reply:
Sorry, what I meant is to use `useRef` because the root element won't change and it does not need to be queried in every render:
```suggestion
  const root = useRef<HTMLElement>(document.querySelector(':root'));
```